File: documentation-examples-java/src/java/dev/typr/foundations/docs/postgresql/RangeTypes.java
package dev.typr.foundations.docs.postgresql;

import dev.typr.foundations.PgType;
import dev.typr.foundations.PgTypes;
import dev.typr.foundations.data.Range;
import dev.typr.foundations.data.RangeBound;

import java.time.LocalDate;

@SuppressWarnings("unused")
public class RangeTypes {
    //start
    PgType<Range<Integer>> intRangeType = PgTypes.int4range;
    PgType<Range<LocalDate>> dateRangeType = PgTypes.daterange;

    // Create ranges with explicit bounds
    Range<Integer> range = Range.int4(
        new RangeBound.Closed<>(1),
        new RangeBound.Closed<>(10)
    );  // [1, 11) after normalization

    // Check containment
    boolean contains = range.contains(5);  // true
    //stop
}

File: documentation-examples-scala/src/scala/dev/typr/foundations/docs/postgresql/RangeTypes.scala
package dev.typr.foundations.docs.postgresql
import dev.typr.scalafoundations.*
import dev.typr.scalafoundations.data.*

import java.time.LocalDate

@SuppressWarnings(Array("unused"))
object RangeTypes:
  //start
  val intRangeType: PgType[Range[Integer]] = PgTypes.int4range
  val dateRangeType: PgType[Range[LocalDate]] = PgTypes.daterange

  // Create ranges with explicit bounds
  val range: Range[Integer] = Range.int4(
    new RangeBound.Closed[Integer](1),
    new RangeBound.Closed[Integer](10)
  ) // [1, 11) after normalization

  // Check containment
  val contains: Boolean = range.contains(5) // true
  //stop
